src/views/LoginView.vue

```diff
@@ -92,6 +92,10 @@ export default {
       walletProvider: null,
       supportEmailAccounts: config.supportEmailAccounts,
 
+      // this is built up in the created() hook and appended to the login URLs
+      //  in the template
+      oAuth2LoginQueryString: '?',
+
       googleLoginUrl: `${config.apiUrl}/oauth2/login/google`,
```

Comment:
Can you make these guys computed instead? Then we don't have to manually add oAuth2LoginQueryString to them in the markup--they'll update (and get cached) automatically after we calculate the QS to add in the `created()` hook.

Reply:
Actually you can just pull that logic out of the `created()` hook completely and make oAuth2LoginQueryString a computed property too and it will trickle down as `destination` and `pendingUserCode` are read from vuex.

Reply:
Good call 👍

Reply:
I ended up [making the login urls a method instead](https://github.com/codex-protocol/web.codex-viewer/pull/49/files#diff-508cbf9449e3e19d1a503582e897cf93R256), since all three were exactly the same aside from the provider name.